crates/utils/src/utils/validation.rs

```diff
@@ -421,23 +432,62 @@ mod tests {
   }
 
   #[test]
-  fn test_valid_actor_name() {
+  fn test_valid_username() {
     let actor_name_max_length = 20;
-    assert!(is_valid_actor_name("Hello_98", actor_name_max_length).is_ok());
-    assert!(is_valid_actor_name("ten", actor_name_max_length).is_ok());
-    assert!(is_valid_actor_name("تجريب", actor_name_max_length).is_ok());
-    assert!(is_valid_actor_name("تجريب_123", actor_name_max_length).is_ok());
-    assert!(is_valid_actor_name("Владимир", actor_name_max_length).is_ok());
+    assert!(is_valid_username("Hello_98", actor_name_max_length).is_ok());
+    assert!(is_valid_username("ten", actor_name_max_length).is_ok());
+    assert!(is_valid_username("تجريب", actor_name_max_length).is_ok());
+    assert!(is_valid_username("تجريب_123", actor_name_max_length).is_ok());
+    assert!(is_valid_username("Владимир", actor_name_max_length).is_ok());
 
     // mixed scripts
-    assert!(is_valid_actor_name("تجريب_abc", actor_name_max_length).is_err());
-    assert!(is_valid_actor_name("Влад_abc", actor_name_max_length).is_err());
+    assert!(is_valid_username("تجريب_abc", actor_name_max_length).is_err());
+    assert!(is_valid_username("Влад_abc", actor_name_max_length).is_err());
     // dash
-    assert!(is_valid_actor_name("Hello-98", actor_name_max_length).is_err());
+    assert!(is_valid_username("Hello-98", actor_name_max_length).is_err());
     // too short
-    assert!(is_valid_actor_name("a", actor_name_max_length).is_err());
+    assert!(is_valid_username("a", actor_name_max_length).is_err());
     // empty
-    assert!(is_valid_actor_name("", actor_name_max_length).is_err());
+    assert!(is_valid_username("", actor_name_max_length).is_err());
+    // newline
+    assert!(is_valid_username(
+      r"Line1
+
+Line3",
+      actor_name_max_length
+    )
+    .is_err());
+  }
+
+  #[test]
```

Comment:
Thx for adding these tests. I know the regex should reject newlines now, but it'd probably be good to add tests for `\n` just in case.